### PR TITLE
fix (PROJ-7707): Filter selected values from options

### PR DESCRIPTION
**Pull request by rbvea**

### CODE REVIEW

**Review by ewalters-politico — CHANGES_REQUESTED** [review]
- I'm getting a Typescript error when I run this locally, but changing "SelectedItem" to "SelectedOption" fixes it
- I tested this change locally and it's causing legos to throw an error when I test it in reggie.  So there may be more changes required on the legos side as well

**Comment by rbvea** [comment]
@ewalters-politico what error did you get? It would help with debugging.

**Comment by ewalters-politico** [comment]
> @ewalters-politico what error did you get? It would help with debugging.

Something to the effect of  `cannot call property 'find' of undefined`.  The stack trace is not helpful, but I think it's referring to GeneratedSelectMulti in  legos, probably the currentOptions or currentlySelectedOptions properties.

**Comment by rbvea** [comment]
@ewalters-politico how do I test this on Reggie? I'm making a small change, and:

- Using yalc to publish `vue-accessible-selects`
- Using yalc to publish `legos-vue`
- Using yalc to add `legos`

Yet I can't see the small change.

**Comment by ewalters-politico** [comment]
> @ewalters-politico how do I test this on Reggie? I'm making a small change, and:
> 
> * Using yalc to publish `vue-accessible-selects`
> * Using yalc to publish `legos-vue3`
> * Using yalc to add `@politico/legos`
> 
> Yet I can't see the small change.

@rbvea , I was able to test local changes by following these steps:
I was able to test local changes by:
1. Make a change in vue-accessible-selects
2. Run `yarn build`
3. Run `yalc publish`
4. Switch to reggie
5. cd to src/client
6. Run `yalc add @politico/vue-accessible-selects`

Legos doesn't include vue-accessible-selects , so it gets installed in reggie on its own, which means that we don't deal with legos at all when testing local changes to VAS.

**Comment by rbvea** [comment]
@ewalters-politico I don't think that's correct that Legos doesn't include VAS. We moved `@politico/search` into Legos and that [uses VAS](https://github.com/politico/legos-vue3/blob/76815643d22ff1746e4a5441780ad717cc97351d/src/components/PoliticoSearch/components/GeneratedSelectMulti.vue#L6) to generate these dropdowns. How do I test updated VAS functionality through legos?

**Comment by ewalters-politico** [comment]
@rbvea , I wasn't specific enough with my above comment - Legos uses VAS, but it does not bundle VAS into the legos package (I just double checked the legos build output to make sure).  This means projects that use legos must install VAS as well as legos.

If you want to test updated VAS functionality through legos only (using the Legos storybook site), it should be the same process that I outlined above for reggie.  One caveat is that Vite based consuming projects [do not play well](https://github.com/vitejs/vite/blob/main/docs/guide/troubleshooting.md#outdated-pre-bundled-deps-when-linking-to-a-local-package) with local dependency project linking via `npm link` or `yalc`. If your newest VAS changes do not show in your legos, delete the `node_modules/.vite` directory in legos and try again.

**Comment by rbvea** [comment]
Thanks @ewalters-politico, I'm not seeing that with my latest commit so would you be able to re-review?

File: src/SelectMulti.vue [hunk 1]
```diff
@@ -83,14 +83,14 @@
 				default: () => null
 			},
 			/**
-             * Determines whether or not typing in the input will filter the provided options.
+			 * Determines whether or not typing in the input will filter the provided options.
 			 * Should be set to false if you are making API calls to fetch options.
```
Comment (tcrite1) [inline src/SelectMulti.vue:87]: @rbvea should it be set to false if you're making api calls to fetch options while typing? So, if I make an initial call to get options, but don't make any subsequent calls based on the user input, I should set this to true?

Comment (tcrite1) [reply src/SelectMulti.vue:87]: also, have you noticed a huge drop in performance when this is set to true?

Comment (rbvea) [reply src/SelectMulti.vue:87]: @tcrite1 yes, when you load options after the initial render, this is the setting to set to `true`. I haven't noticed a drop in performance, have you?

Comment (tcrite1) [reply src/SelectMulti.vue:87]: @rbvea when this is set to true, I have noticed a drop in performance. It's not related to this change though. Also, our list is like 800 options :-/

Comment (rbvea) [reply src/SelectMulti.vue:87]: I'm seeing that y'all are using it for the [SearchFilters](https://github.com/politico/delta/blob/develop/src/client/src/components/search/SearchFilters.vue). Is it a "dumb" implementation — in that it's derived from the facets returned from Sail? If so then I don't think the flag is needed. 

Over in Pro we use it whenever :options can vary. In politico/search, there's a concept of "[currentOptions](https://github.com/politico/legos-vue3/blob/0a831317e308267ad9e872043d29f79117d93148/src/components/PoliticoSearch/components/GeneratedSelectMulti.vue#L76-L84)" pared with "clientEscapeHatches", which [fetches the dropdown options from autocomplete](https://github.com/politico/reggie/blob/16b25b819f466e359d7374d693632b003d308f56/src/client/src/components/search/filters/clientEscapeHatchesMapping.ts#L27-L30).


Only if :options are populated by autocomplete, then the prop should be set to true.

Comment (tcrite1) [reply src/SelectMulti.vue:87]: It's working okay for the search filters. it's this one where I'm seeing the issue?

https://github.com/politico/delta/blob/develop/src/client/src/components/shared/UserShare.vue

**Pull request merged**